### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by ifiokjr** [ifiokjr/remirror#23]
## Description

Currently, the individual package documentation is lacking. 

Each package should include the following

- [x] Badges for build size, latest version, downloads / month, open pull requests by label, open issues by label, close prs by label, license, stars
- [x] Integrate [**lgtm**](https://lgtm.com/)
- [x] Logo on all readme.md pages
- [ ] Code examples for each package
- [ ] Include `/packages` with each package documented as a path in docs.remirror.org/packages/<package-name>
- [ ] Getting started guide i.e. by building an editor example
- [ ] Build an example extension guide

Example usage for:

- [ ] Twitter UI
- [ ] GitHub UI
- [ ] Floating Menu

### User requested documentation

- [ ] [RenderTree] document how to configure RenderTree if Remirror has user-defined extensions like new marks or nodes #66
- [ ] [Remirror] document how to hook into Remirror events / prosemirror events 
  - Is the way to do this by configuring proseMirror [`EditorProps`](https://github.com/ifiokjr/remirror/blob/master/support/types/prosemirror-view/index.d.ts#L406) on `<Remirror>` instance, i.e. something like `<Remirror handleTripleClick={(view, pos, event) => { someCallback() }} />`?
  - Or by configuring a Remirror extension - see [@remirror/ui-twitter/src/components/ui-twitter.tsx#L84](https://github.com/ifiokjr/remirror/blob/master/@remirror/ui-twitter/src/components/ui-twitter.tsx#L84)
- [ ] [tutorial] demonstrate how to set up Remirror editor that can toggle between markdown editing and toolbar editing just like https://prosemirror.net/examples/markdown/

**Comment by 100ideas** [comment]
I'd like to contribute to the documentation as I teach myself prosemirror & attempt to integrate remirror into one of my own projects.

I propose using this issue (or starting a new one) to maintain a todo-list of documentation tasks & suggestions. In particular I would like a place where I can list specific bits of documentation I needed but couldn't find while using remirror. I'll start the list in my next message. If the list proves useful, we can promote it to a new issue and I'll keep the todo list updated in first message, and we can use the comments for discussion. An example of the discussion content I might post is follows.

The rest of this message is a cross-post of a response I made to you in your [thread on the prosemirror forums](https://discuss.prosemirror.net/t/using-with-react/904/15?u=100ideas).

---
> https://discuss.prosemirror.net/t/using-with-react/904/15?u=100ideas:
>
> I took a small step into the codebase last night by trying to get the [simplest example in the docs](https://github.com/ifiokjr/remirror/blob/master/docs/examples/document-model.tsx) working in my app's react-styleguidist component library.
> 
> Copy-pastying the example code basically worked. [Here's a gist of the little editor component I was playing around with](https://gist.github.com/100ideas/9ba3aa4fffc5d44aebfbdcf1b437f286) (it should be possible to put `package.json` in root and everything else in `src` then hit `yarn; yarn run styles` to get it to run locally).
> 
> One area I had to puzzle through was how to properly initialize the `<RenderTree />` json viewer. I initially copied-and-pasted the output of the `RenderTree` component as json back into my source code so I could initialize the editor with more content. However, `RenderTree` didn't know how to interpret `bold` or `italic` marks that the stock `remirror` editor generated. I eventually figured out how to [pass in a modifed `markMap`](https://gist.github.com/100ideas/9ba3aa4fffc5d44aebfbdcf1b437f286#file-textblockeditor-jsx-L137). I didn't try to hard to read all the documentation since I assumed it was in flux, so I may have missed the details about that. otherwise that info would be good to add to the docs.
> 
> I'd be happy to help with that.
> 
> Then I got stuck - I want to add a way for the user to to toggle the `<EditorLayout>` and/or `<RenderTree>` between menubar edit mode & direct markdown entry mode, a la https://prosemirror.net/examples/markdown/. 
> 
> As a first step I just wanted to show/hide various components by triple-clicking the RenderTree or executing a special keystroke.
> 
> I saw that there are type definitions for prosemirror config props to do that sort of thing in 
> [remirror/support/types/prosemirror-view/index.d.ts#L406](https://github.com/ifiokjr/remirror/blob/master/support/types/prosemirror-view/index.d.ts#L406), but I couldn't figure out where or how to pass in props to `<RenderTree>` - **could you give me a hint or two?**
> 
> ```js 
> // remirror/blob/master/support/types/prosemirror-view/index.d.ts#L406
> /**
>  * Props are configuration values that can be passed to an editor view
>  * or included in a plugin. This interface lists the supported props.
>  *
>  * The various event-handling functions may all return `true` to
>  * indicate that they handled the given event. The view will then take
>  * care to call `preventDefault` on the event, except with
>  * `handleDOMEvents`, where the handler itself is responsible for that.
>  *
>  * How a prop is resolved depends on the prop. Handler functions are
>  * called one at a time, starting with the base props and then
>  * searching through the plugins (in order of appearance) until one of
>  * them returns true. For some props, the first plugin that yields a
>  * value gets precedence.
>  */
> export interface EditorProps<S extends Schema = any> {
>   /**
>    * Can be an object mapping DOM event type names to functions that
>    * handle them. Such functions will be called before any handling
>    * ProseMirror does of events fired on the editable DOM element.
>    * Contrary to the other event handling props, when returning true
>    * from such a function, you are responsible for calling
>    * `preventDefault` yourself (or not, if you want to allow the
>    * default behavior).
>    */
> 
> // skipping down a few lines...
> 
>    /**
>    * Called when the editor is triple-clicked, after `handleTripleClickOn`.
>    */
> handleTripleClick?: ((view: EditorView<S>, pos: number, event: MouseEvent) => boolean) | null;
> ```
> 
> p.s. happy to move this convo over to your repo moving forward, just let me know.

**Comment by 100ideas** [comment]
- [ ] [RenderTree] document how to configure RenderTree if Remirror has user-defined extensions like new marks or nodes #66

- [ ] [Remirror] document how to hook into Remirror events / prosemirror events 

  - Is the way to do this by configuring proseMirror [`EditorProps`](https://github.com/ifiokjr/remirror/blob/master/support/types/prosemirror-view/index.d.ts#L406) on `<Remirror>` instance, i.e. something like `<Remirror handleTripleClick={(view, pos, event) => { someCallback() }} />`?
  - Or by configuring a Remirror extension - see [@remirror/ui-twitter/src/components/ui-twitter.tsx#L84](https://github.com/ifiokjr/remirror/blob/master/@remirror/ui-twitter/src/components/ui-twitter.tsx#L84)

- [ ] [tutorial] demonstrate how to set up Remirror editor that can toggle between markdown editing and toolbar editing just like https://prosemirror.net/examples/markdown/

**Comment by charlex** [comment]
Small piece of feedback, that could be an easy todo:

It seems like headers in the docs are a bit inconsistent. It seems like we'd want to make sure each page has a similar title just to help users navigate and keep things looking clean.

**For example: a page with a large title (h1)**

![image](https://user-images.githubusercontent.com/1542740/61584843-89d0df00-ab3e-11e9-9e53-68a707f2c048.png)


**For example: a page without a title**

![image](https://user-images.githubusercontent.com/1542740/61584827-40808f80-ab3e-11e9-86ae-e86e3e4e37de.png)

**Comment by charlex** [comment]
Another addition to the todo list since I can't seem to find any other relevant tickets:

I am right now in the midst of trying to figure out how to implement an html-only (no markdown) controlled editor. I went to https://docs.remirror.org/guides/controlled-editor and noticed that it seems to just be placeholder. This would be really helpful, as I'm sure it will increasingly become an important usecase for complex apps.

@ifiokjr Do you have any example source code that could aid in the writing of this guide? I have been trying to follow [this](https://github.com/ifiokjr/remirror/tree/master/@remirror/editor-markdown/src/editor.tsx), but with the addition of the markdown code, its a bit hard to follow. I'm sure i'll be able to get this done, but it would be faster to have any guidance you can offer.

**Comment by ifiokjr** [comment]
> @ifiokjr Do you have any example source code that could aid in the writing of this guide? I have been trying to follow [this](https://github.com/ifiokjr/remirror/tree/master/@remirror/editor-markdown/src/editor.tsx) [](/ifiokjr/remirror/tree/HEAD@{2019-07-21T03:48:47Z}/@remirror/editor-markdown/src/editor.tsx), but with the addition of the markdown code, its a bit hard to follow. I'm sure i'll be able to get this done, but it would be faster to have any guidance you can offer.

This will be very appreciated 👍  

Let me know how I can help.

Also, I've actually been meaning to revisit that markdown editor code. It's completely unfinished so this is a timely reminder to get it into shape.

I'll probably be able to provide better assistance once that's in progress. I created the controlled editor in response to a feature request but I haven't had time to dive into the intricacies, caveats and benefits of using it.

**Comment by charlex** [comment]
Thanks @ifiokjr! For me, the big benefit is being able to, without remounting the editor, swap the content out on the fly.

This allows me to have an array of content, and when the user selects an item in the array, the editor will then be populated with that content. This array could be stories from an RSS feed, or different pieces of available content, etc, etc.

**Comment by ifiokjr** [comment]
@charlex I'll get started on this probably towards the backend of this week.
